src/hooks/auth/useRegister.ts

```diff
@@ -3,30 +3,31 @@ import { useNavigate } from 'react-router-dom'
 import { registerThunk } from 'store/thunk'
 import { REGISTRATION_FORM_INITIAL_VALUES } from 'constants/auth/registration'
 import { PUBLIC_PAGES } from 'constants/pages'
-import { ROLES } from 'constants/profile'
 import { isRejectedAction } from 'utils/store'
 import { useAppDispatch } from '../useAppDispatch'
 
-export const useRegister = () => {
+export const useRegister = (email: string, token: string) => {
   const navigate = useNavigate()
   const dispatch = useAppDispatch()
 
   return useCallback(
     async (values: typeof REGISTRATION_FORM_INITIAL_VALUES) => {
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      const { confirmPassword, ...profileData } = values
+      const { newPassword, confirmPassword } = values
 
       const res = await dispatch(
         registerThunk({
-          ...profileData,
-          role: ROLES.general,
+          newPassword,
+          confirmPassword,
+          email,
+          token,
```

Comment:
Get url params here by useQueryParams rather than getting them outside and passing it to this hook.